src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -86,24 +87,31 @@ template Batch
             oHoldingCid <-
               T.mapA (\tCid -> fromOptional tCid <$> undiscloseT actors tCid) oSettledHoldingCid
             pure (oHoldingCid, usedTypeRepOpts)
-
         -- execute instructions
         (oHoldingCids, oUsedTypeReps) <- unzip <$> mapA settleInstruction orderedInstructions
-
         -- consistency check
         forA_ (groupOn fst . catOptionals . concat $ oUsedTypeReps) \ts ->
           assertMsg "Allocated holdings of the same instrument must have the same template type rep"
-            $ size (fromList ts) == 1
-
+            $ S.size (S.fromList ts) == 1
         pure $ catOptionals oHoldingCids
+      cancel Batch.Cancel{actors} = do
+        assertMsg "actors must contain requestors" $ requestors `S.isSubsetOf` actors
+        let
+          cancelInstruction instruction = do
+            instructionCid <- fst <$> (fetchByKey @Instruction instruction)
+            exercise (toInterfaceContractId @Instruction.I instructionCid) Instruction.Cancel with
+              actors
+            pure ()
+        -- cancel instructions
+        forA_ (map (buildKey . snd) routedStepsWithInstructionId) cancelInstruction
```

Comment:
```suggestion
        forA_ routedStepsWithInstructionId $ cancelInstruction . buildKey . snd
```

Probably something like this also works

Reply:
yes, I changed to `mapA_ cancelInstruction . buildKey . snd $ routedStepsWithInstructionId`